utils/field.c

```diff
@@ -49,6 +49,17 @@ int print_empty_field(struct list_head *output_fields, int space)
 	return 1;
 }
 
+void print_addr(struct field_data *fd)
+{
+	/* uftrace records (truncated) 48-bit addresses */
+	int width = sizeof(long) == 4 ? 8 : 12;
+
+	if (fd->addr == 0)  /* for EVENT or LOST record */
+		fd->print("%*s", width, "");
+	else
+		fd->print("%*lx", width, fd->addr);
+}
```

Comment:
I understand your intention to have a common function to deal with different address sizes.  But I'm not sure it's worth specializing the address handling.

Maybe we could add a function to get a portion of the effective address from a given address just before printing.  Also using `sizeof(long)` is wrong since it didn't consider cross-platform data analysis.  I know it's not your fault, but I'd like to change it to use size of address in the data file - something like `data_is_lp64()`.